### ISSUE TITLE
Ruby Cucumber Total number of tests not reflecting the actual nr of tests  - BDD statistics

### ISSUE THREAD

**Issue by skaliber** [reportportal/reportportal#202]
Since I update reportportal to latest version I have an issue with my total nr of tests that reportportal is counting for each launch.

I'm using agent ruby to sent the results into RP and I'm running parallel docker containers 
 so the end of my docker command looks like this:

`-f pretty -f ReportPortal::Cucumber::Formatter -o 'output/test_report.txt'`

when I go to http://rp-host/ui/#{project}/launches/all?page.page=1&page.size=50
I can see the launches running but the total number of tests / launch is wrong

and if I go to the last child of the launch to see the tests I can see all the tests that has ben executed but this doesn't reflect in any of the stats that reportportal is creating after each launch.

You can see in my attached screenshots at the root level It shows 4 tests and at the last leaf level I can see all of them.

we're currently on 3.0.5 for the UI, 3.0.1 for the API, 3.0.0 for the gateway, 3.0.2 for auth

![screen shot 2017-08-02 at 12 55 38](https://user-images.githubusercontent.com/1927769/28872676-28d5a88a-7782-11e7-8c70-82257ee0332e.png)
![screen shot 2017-08-02 at 12 56 00](https://user-images.githubusercontent.com/1927769/28872677-28d63a70-7782-11e7-8caf-5339b5504ffa.png)
![screen shot 2017-08-02 at 12 56 07](https://user-images.githubusercontent.com/1927769/28872680-28d89248-7782-11e7-9c4e-b36f7419a2af.png)
![screen shot 2017-08-02 at 12 56 14](https://user-images.githubusercontent.com/1927769/28872678-28d6ac9e-7782-11e7-8122-ce3fbd5af2d8.png)
![screen shot 2017-08-02 at 12 56 20](https://user-images.githubusercontent.com/1927769/28872679-28d7f266-7782-11e7-95fb-19d65ae17b5c.png)
![screen shot 2017-08-02 at 12 56 26](https://user-images.githubusercontent.com/1927769/28872681-28d91e20-7782-11e7-86f9-02974ac7a402.png)

**Comment by skaliber** [comment]
I would like to mention that when I run sequentially all is good is just when I run in parallel , I guess something is not handling the threads on your side

**Comment by avarabyeu** [comment]
@aliaksandr-trush could you please have a look?

**Comment by skaliber** [comment]
if helps I've changed the component agent to use `YARPC::Cucumber::Formatter `and the issue is still persistent

**Comment by skaliber** [comment]
as per slack discussion I was asked if I can provide a a minimum reproducible example
here is our project just with a sample test that I was using today to test the issue 
It contains the same structure and same gems as we have

https://github.com/skaliber/reportportal_sample_project.git

Thank you!

**Comment by pictureb00k** [comment]
@skaliber
In new version we added different types of statistics calculation, and update scripts change statistics calculation type to new default value.
If i understood from slack chat, you have not used correct update to another version and your type has not changed.
To verify it, please, go to API -> project-controller -> Get information about project -> enter your project name and find field statisticCalculationStrategy.
If it is "TEST_BASED", it doesn't include step level test cases and here is a problem.

**Comment by DzmitryHumianiuk** [comment]
@skaliber issue fixed?

close as duplicate of https://github.com/reportportal/reportportal/issues/83
